module/VuFind/src/VuFind/Http/GuzzleService.php

```diff
@@ -90,6 +89,19 @@ public function __construct(array $config)
         }
     }
 
+    /**
+     * Return a generic PSR-compliant client.
+     *
+     * @param ?string $url     Target URL (required for proper proxy setup for non-local addresses)
+     * @param ?float  $timeout Request timeout in seconds (overrides configuration)
+     *
+     * @return ClientInterface
+     */
+    public function createClient(?string $url = null, ?float $timeout = null): \Psr\Http\Client\ClientInterface
+    {
+        return new \GuzzleHttp\Client($this->getGuzzleConfig($url, $timeout));
```

Comment:
To reduce redundancy of logic, I'd suggest:


```suggestion
        return $this->createGuzzleClient($url, $timeout);
```